### PR TITLE
Feature/improved custom fee wlt 5351

### PR DESCRIPTION
**Pull request by everdimension**

### CODE REVIEW

File: src/ui/pages/SendTransaction/NetworkFee/NetworkFeeDialog.tsx [hunk 1]
```diff
@@ -112,15 +113,22 @@ function setFormValue(form: HTMLFormElement, name: string, value: unknown) {
...
+      event.currentTarget.setCustomValidity('');
+    }
+  };
+
+const setGasPriceValidationMessage = setPatternValidity(
+  'Gas Price value must be a positive number'
+);
+
+const setGasLimitValidationMessage = setPatternValidity(
+  'Gas Limit value must be a natural number'
```
Comment (zerts) [inline src/ui/pages/SendTransaction/NetworkFee/NetworkFeeDialog.tsx:130]: Not sure about the "Natural number" term. Is it indeed correct, but is it a well-known? :)

File: src/modules/ethereum/transactions/gasPrices/estimateNetworkFee.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@ import type { IncomingTransaction } from '../../types/IncomingTransaction';
 import { resolveChainId } from '../resolveChainId';
 import type { ChainId } from '../ChainId';
 import { estimateFee } from './eip1559/estimateFee';
-import { getEip1559Base } from './eip1559/getEip1559Base';
```
Comment (zerts) [inline src/modules/ethereum/transactions/gasPrices/estimateNetworkFee.ts:6]: `getEip1559Base` can be removed completely

Comment (zerts) [reply src/modules/ethereum/transactions/gasPrices/estimateNetworkFee.ts:6]: And I am very happy about it :))

File: src/ui/pages/SendTransaction/NetworkFee/NetworkFeeDialog.tsx (hunk 1 above)
Comment (everdimension) [reply src/ui/pages/SendTransaction/NetworkFee/NetworkFeeDialog.tsx:130]: The alternative is "integer" which also seems technical, so i'm not sure :)

File: src/ui/pages/SendTransaction/TransactionConfiguration/useTransactionFee.ts [hunk 3]
```diff
@@ -236,6 +247,7 @@ export function useTransactionFee({
   );
 
   const feeEstimation = feeEstimationQuery.data?.feeEstimation;
+
```
Comment (zerts) [inline src/ui/pages/SendTransaction/TransactionConfiguration/useTransactionFee.ts:250]: delete this extra line?

**Review by zerts — APPROVED** [review]

**Pull request merged**